libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -275,8 +277,33 @@ export class CheckoutService {
     await this.cartManager.updateFreshCart(cart.id, version, {
       stripeSubscriptionId: subscription.id,
     });
-    const updatedVersion = version + 1;
 
+    const updateCartVersion = version + 1;
+
+    // Process the Stripe payment, and trigger the front end to poll for the cart status
+    this.processStripePayment(
+      cart,
+      updateCartVersion,
+      subscription,
+      confirmationTokenId,
+      customer,
+      uid
+    ).catch((error) => {
+      console.error(error);
+      this.cartManager.finishErrorCart(cart.id, {
+        errorReasonId: CartErrorReasonId.Unknown,
+      });
+    });
+  }
+
+  async processStripePayment(
+    cart: ResultCart,
+    version: number,
+    subscription: StripeSubscription,
+    confirmationTokenId: string,
+    customer: StripeCustomer,
+    uid: string
+  ) {
```

Comment:
Why did this become a separate function? It looks like it's only used by payWithStripe?